dialog/file.go

```diff
@@ -50,28 +107,34 @@ func (f *fileDialog) makeUI() fyne.CanvasObject {
 	}
 
 	label := "Open"
-	if f.save {
+	if f.file.save {
 		label = "Save"
 	}
 	f.open = widget.NewButton(label, func() {
-		if f.callback == nil {
+		if f.file.callback == nil {
 			f.win.Hide()
+			if f.file.onClosedCallback != nil {
```

Comment:
Should this be conditional on `file.callback` not being nil?

Reply:
file.callback is the read/write closer callback ... file.onClosedCallback is the onClosed callbacks ... They are unrelated as far as I can tell

Reply:
Oh I see you were actually referring to code above. In that space we only hide immediately if there is no file.callback ... so we wouldn't want to call the onClosedCallback until the dialog is closed... in this case when file.callback is not nil, we go through things like validation that may keep the dialog open still - if validation passes we call the Hide() and onClosedCallbacks later in the method...

So long story short - I think this is appropriate.

Reply:
Thanks for the explanation, but I don't quite follow - there is a return here so I don't see where it falls through for the extra processes?

Reply:
Oh I see what you mean now sorry